### PR TITLE
Refactor frontend for API server

### LINKED ISSUE #48: Refactor Finage & CMC calls to use API server

### PR DESCRIPTION
**Pull request by danielfamiyeh**
fixes #48

### CODE REVIEW

File: frontend/pages/bonds/index.vue [hunk 1]
```diff
@@ -1,7 +1,17 @@
 <template>
   <div v-if="loading">
-    <div class="container d-flex justify-content-around w-50 pt-5 vh-100 text-center">
-      <div class="loading-logo mt-5" role="status"/>
+    <div
+      class="
+        container
```
Comment (pixelomo) [inline frontend/pages/bonds/index.vue:5]: this is prettier formatting please remove

File: frontend/nuxt.config.js [hunk 2]
```diff
@@ -65,55 +88,52 @@ export default {
...
 
   // Modules (https://go.nuxtjs.dev/config-modules)
   modules: [
     // https://go.nuxtjs.dev/bootstrap
-    'bootstrap-vue/nuxt',
-    '@nuxtjs/proxy',
-    '@nuxtjs/style-resources',
-    '@nuxtjs/axios',
-    '@nuxtjs/google-adsense',
+    "bootstrap-vue/nuxt",
```
Comment (pixelomo) [inline frontend/nuxt.config.js:128]: please undo prettier formatting

File: frontend/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (pixelomo) [inline frontend/.prettierrc:1]: please remove prettierrc

File: frontend/pages/bonds/_symbol.vue [hunk 4]
```diff
@@ -1,10 +1,21 @@
 <template>
   <div v-if="loading">
-    <div class="container d-flex justify-content-around w-50 pt-5 vh-100 text-center">
-      <div class="loading-logo mt-5" role="status"/>
+    <div
+      class="
+        container
```
Comment (pixelomo) [inline frontend/pages/bonds/_symbol.vue:5]: please undo prettier formatting

File: frontend/pages/commodities/_symbol.vue [hunk 5]
```diff
@@ -1,10 +1,21 @@
 <template>
   <div v-if="loading">
-    <div class="container d-flex justify-content-around w-50 pt-5 vh-100 text-center">
-      <div class="loading-logo mt-5" role="status"/>
+    <div
+      class="
+        container
```
Comment (pixelomo) [inline frontend/pages/commodities/_symbol.vue:5]: please undo prettier formatting

File: frontend/pages/commodities/index.vue [hunk 6]
```diff
@@ -1,15 +1,22 @@
 <template>
   <div v-if="loading">
-    <div class="container d-flex justify-content-around w-50 pt-5 vh-100 text-center">
-      <div class="loading-logo mt-5" role="status"/>
+    <div
+      class="
+        container
```
Comment (pixelomo) [inline frontend/pages/commodities/index.vue:5]: please undo prettier formatting

File: frontend/pages/commodities/index.vue [hunk 7]
```diff
@@ -40,86 +47,97 @@
...
+  components: {
+    IndexList: () => import("./../../components/IndexList"),
+  },
+  data() {
+    return {
+      finageApiKey: process.env.finageApiKey,
+      tradingEconKey: process.env.tradingEconKey,
+      loading: true,
+      commodities,
+      view: "list",
```
Comment (pixelomo) [inline frontend/pages/commodities/index.vue:63]: please undo prettier formatting

File: frontend/pages/cryptocurrency/_symbol.vue [hunk 8]
```diff
@@ -1,6 +1,16 @@
 <template>
   <div v-if="loading">
-    <div class="container d-flex justify-content-around w-50 pt-5 vh-100 text-center">
+    <div
+      class="
+        container
+        d-flex
```
Comment (pixelomo) [inline frontend/pages/cryptocurrency/_symbol.vue:6]: please undo prettier formatting

File: frontend/pages/currencies/_symbol.vue [hunk 9]
```diff
@@ -1,10 +1,21 @@
 <template>
   <div v-if="loading">
-    <div class="container d-flex justify-content-around w-50 pt-5 vh-100 text-center">
-      <div class="loading-logo mt-5" role="status"/>
+    <div
+      class="
+        container
```
Comment (pixelomo) [inline frontend/pages/currencies/_symbol.vue:5]: please undo prettier formatting

File: frontend/pages/currencies/_symbol.vue [hunk 10]
```diff
@@ -22,267 +33,299 @@
...
+  data() {
+    return {
+      finageApiKey: process.env.finageApiKey,
+      liveApiUrl: process.env.liveApiUrl,
+      item: {
+        name: "",
+        price: 0,
+        icon: "",
+      },
+      profile: {},
```
Comment (pixelomo) [inline frontend/pages/currencies/_symbol.vue:52]: please undo prettier formatting

File: frontend/pages/stocks/_symbol.vue [hunk 11]
```diff
@@ -23,213 +34,302 @@
 </template>
 
 <script>
-import {stocks} from "../../market.js";
-import Item from '~/components/Item.vue'
+import { stocks } from "../../market.js";
+import { useQuery } from "@/services/graphql.js";
+b;
```
Comment (pixelomo) [inline frontend/pages/stocks/_symbol.vue:39]: I assume b; is a typo?

**Review by pixelomo — CHANGES_REQUESTED** [review]
Prettier formatting is still present across all files, please remove this so that I can review and merge. I'm also not seeing any changes for data fetching on the homepage in /pages/index.vue

**Comment by danielfamiyeh** [comment]
Closing as it's easier to create a new PR based off of main, than make changes to merge into main

**Pull request closed without merging**